src/components/shared/Entity/DetailsForm.tsx

```diff
@@ -89,20 +99,19 @@ function DetailsForm({
     >(formStateStoreName, (state) => state.resetState);
 
     useEffect(() => {
-        if (connectorID) {
-            setDetails({
-                data: {
-                    entityName: '',
-                    connectorImage: {
-                        id: connectorID,
-                        iconPath: '',
-                        connectorId: '',
-                        imagePath: '',
-                    },
-                },
+        if (connectorID && hasLength(connectorTags)) {
+            connectorTags.forEach((connector) => {
+                if (connector.connector_tags[0].id === connectorID) {
+                    setDetails({
+                        data: {
+                            entityName: '',
+                            connectorImage: getConnectorImageDetails(connector),
+                        },
+                    });
+                }
```

Comment:
Now we'll wait until we have connectorTags and _then_ set the prefilled connector. This allows us to populate all the props we need.